CDCgov/prime-reportstream
# Victor/7932 UP accepts bulk FHIR bundles

This PR allows the Universal Pipeline to accept and ingest bulk FHIR bundles through the reports API endpoint. To be able to send bulk FHIR bundles the new content type application/fhir+ndjson had to be added. Ndjson stands for New line Delimited JSON and allows us to send multiple FHIR bundles new line delimited in one message. Previous smoke tests using FHIR data had to be modified to remove any new lines that would break the tests.

Test Steps:

1. Send a POST request to send a bulk FHIR report. Sample request: 
[Tests.postman_collection.txt](https://github.com/CDCgov/prime-reportstream/files/10572145/Tests.postman_collection.txt)


2. Verify that the request was accepted

3. Call the submission history API to verify the reports were delivered: http://localhost:7071/api/waters/report/{{reportId}}/history
4. Optional: Check sftp folder to verify that the HL7 message get routed correctly


## Changes
- Updated the FHIR format in both Sender.kt and Report.kt to support bulk FHIR bundles (application/fhir+ndjson)
- Added a new FHIRReader.kt class that takes in a raw fhir+ndjson message and returns a list of bundles
- Updated ELRReceiver.kt to be able to support sending and receving bulk FHIR bundles
- Fixed a bug on FHIRConverter.kt that didn't allow for multiple bundles to be sent to Routing
- Fixed a bug on SubmissionHistory.kt that would create the same destination multiple times in the API response
- Updated unit tests to use new fhir+ndjson message format

## Checklist

### Testing

## Linked Issues
- Fixes #7932
